### PR TITLE
feat(deploy-tooling): create transport for undeployment

### PR DESCRIPTION
**Pull request by cianmSAP**
#1112

### CODE REVIEW

File: packages/deploy-tooling/README.md [hunk 1]
```diff
@@ -101,6 +101,8 @@ Options:
   --username                           ABAP Service username
   --password                           ABAP Service password
   --transport <transport-request>      Transport number to record the change in the ABAP system
+  --create-transport                   Create a transport request during deployment
+  --package <abap-package>             Package name for deploy target ABAP system (only required when --create-transport is used)
```
Comment (tobiasqueck) [inline packages/deploy-tooling/README.md:105]: The `package` shouldn't be required for undeploy. The deployed app is already part of a package and undeployment doesn't need to know which on (I think).

File: packages/deploy-tooling/src/cli/index.ts [hunk 2]
```diff
@@ -43,7 +43,11 @@ export function createCommand(name: 'deploy' | 'undeploy'): Command {
                 'Read ABAP cloud service properties from environment variables or .env file'
             ).conflicts(['cloudServiceKey', 'destination'])
         )
+        .option('--package <abap-package>', 'Package name for deploy target ABAP system')
```
Comment (tobiasqueck) [inline packages/deploy-tooling/src/cli/index.ts:46]: Same as above, I don't think it is used.

**Review by tobiasqueck — CHANGES_REQUESTED** [review]

File: packages/deploy-tooling/README.md (hunk 1 above)
Comment (cianmSAP) [reply packages/deploy-tooling/README.md:105]: The `package` is a required param for `createTransportRequest` 
https://github.com/SAP/open-ux-tools/blob/84579be8d6780b5133524cc66b3aab1dc68a2d83/packages/axios-extension/src/abap/adt-catalog/services/transportrequest-service.ts#L7-L12

Comment (tobiasqueck) [reply packages/deploy-tooling/README.md:105]: @slin-sap you are the ADT expert here, is it an issue with the type or is a package name really required for creating a transport to undeploy an existing BSP?

Comment (tobiasqueck) [reply packages/deploy-tooling/README.md:105]: Weird but ok, retracting my comment then.

**Review by tobiasqueck — APPROVED** [review]
* changes are consistent with the deploy command
* code is easy to read and maintain
* test cover the new functionality
* no manual tests but no doubts about it working based on the previous functionality

**Pull request merged**